hive/src/main/java/com/netease/arctic/hive/utils/HiveTableUtil.java

```diff
@@ -134,4 +143,138 @@ public static StorageDescriptor storageDescriptor(
     storageDescriptor.setSerdeInfo(serDeInfo);
     return storageDescriptor;
   }
+
+  public List<Partition> getHiveAllPartitions(HMSClient hiveClient, TableIdentifier tableIdentifier) {
+    try {
+      return hiveClient.run(client ->
+          client.listPartitions(tableIdentifier.getDatabase(), tableIdentifier.getTableName(), Short.MAX_VALUE));
+    } catch (NoSuchObjectException e) {
+      throw new NoSuchTableException(e, "Hive table does not exist: %s", tableIdentifier.getTableName());
+    } catch (TException e) {
+      throw new RuntimeException("Failed to get partitions " + tableIdentifier.getTableName(), e);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to listPartitions", e);
+    }
+  }
+
+  public static List<String> getHivePartitionNames(HMSClient hiveClient, TableIdentifier tableIdentifier) {
```

Comment:
recommend to move it to HivePartitionUtil

Reply:
It has been moved